### PR TITLE
WIP: Fetching triples from a JSON-LD document

### PR DESCRIPTION
**Pull request by angelo-v**
I added a failing test for a problem I observed fetching data from JSON-LD documents. The PR does not include a fix yet. Reviews, if the test actually uncovers a bug that needs fixing are welcome. The same is true for ideas how to fix.

### CODE REVIEW

File: tests/unit/fetcher-jsonld-test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+const {expect} = chai
+
+describe('Given a JSON-LD resource', () => {
+
+    const uri = "http://localhost/jsonld#it"
+
+    beforeEach(() => {
+        const docContents = `
+        {
+            "@id": ${uri},
```
Comment (josephguillaume) [inline tests/unit/fetcher-jsonld-test.js:15]: There's a small typo here:
```suggestion
            "@id": "${uri}",
```
I haven't run the test, but if it passes now, then maybe the issue you were facing was #408? 
Note also that another bug caused jsonld errors to be suppressed: #435 

I've added some annotations to the relevant commit https://github.com/linkeddata/rdflib.js/commit/16daad7d68d805aa29e8e01bb4a16d77b28b6afe#r41313390

Comment (josephguillaume) [reply tests/unit/fetcher-jsonld-test.js:15]: Or is there a separate issue related to fetching, like in https://github.com/linkeddata/rdflib.js/issues/343?

Comment (angelo-v) [reply tests/unit/fetcher-jsonld-test.js:15]: Good catch! Fixed the syntax. But test is still failing for the same reason. I think the issue is in the fetching / response handling.

There is just no default handler for JSON-LD?

```
const defaultHandlers = {
  RDFXMLHandler, XHTMLHandler, XMLHandler, HTMLHandler, TextHandler, N3Handler
}
```

**Comment by michielbdejong** [comment]
This also makes NSS fail the content negotiation tests I'm adding in https://github.com/solid/solid-crud-tests/commit/9d4928b8abedb0f791b0f2a92a6a9a9e0d4eb774

**Comment by michielbdejong** [comment]
I'll see what I can do to try to fix this bug.

**Comment by michielbdejong** [comment]
Indeed it cannot find a JSON-LD handler, these are my debug statements from `handlerForContentType`:
```
handle response application/ld+json
checking pattern /application\/rdf\+xml/ null
checking pattern /application\/xhtml/ null
checking pattern /(text|application)\/(.*)xml/ null
checking pattern /text\/html/ null
checking pattern /text\/plain/ null
checking pattern /(application|text)\/(x-)?(rdf\+)?(n3|turtle)/ null
no handler
```

**Comment by michielbdejong** [comment]
Continued in https://github.com/linkeddata/rdflib.js/pull/439

**Pull request closed without merging**